docs/CLI.md

```diff
@@ -1692,6 +1692,16 @@ Path to Moodle
 * Is negatable: no
 * Default: `'.'`
 
+#### `--max-warnings`
+
+Number of warnings to trigger nonzero exit code - default: -1
+
+* Accept value: yes
+* Is value required: yes
```

Comment:
I think that "Is value required" means that, if you use the option, a value needs to be specified (the oposite of other options, like `--verbose` or `--fail-on-warning` (in phpunit) that are just options (switches) without value allowed.

Surely it could be expressed another way, but as far as it's auto-generated I'm not sure if applying for more changes (we already do some regexps) is a good idea.

CIao :-)

Reply:
Understood... and it makes sense.
Apologies for the noise and thanks for your time 💪 .

Matteo

Reply:
Nono, it’s true that is confusing, because there are the “accept value”, “require value” and “default value” specs for every option.And the combinations of them aren’t clear. In fact I don’t like that default -1 at all, it should be kept behind the scenes (because it’s the value that is used INTERNALLY when the option is not specified. If specified, then the default isn’t applied ever, because a value MUST be provided,

So it is a good point, just I prefer it to continue being auto.generated than maintaining it manually.

Reply:
(Will try to look if we can delete those “fake” defaults. Maybe then it’s clear. But not in this issue